Library/Homebrew/cmd/list.rb

```diff
@@ -109,26 +109,33 @@ def list
       ls_args << "-r" if args.r?
       ls_args << "-t" if args.t?
 
-      if HOMEBREW_CELLAR.exist? && HOMEBREW_CELLAR.children.any?
+      if !args.cask? && HOMEBREW_CELLAR.exist? && HOMEBREW_CELLAR.children.any?
         ohai "Formulae" if $stdout.tty? && !args.formula?
         safe_system "ls", *ls_args, HOMEBREW_CELLAR
       end
-
       if !args.formula? && Cask::Caskroom.casks.any?
-        if $stdout.tty?
+        if $stdout.tty? && !args.cask?
           puts
           ohai "Casks"
         end
         safe_system "ls", *ls_args, Cask::Caskroom.path
       end
-    elsif args.verbose? || !$stdout.tty?
-      system_command! "find", args:         args.named.to_default_kegs.map(&:to_s) + %w[-not -type d -print],
-                              print_stdout: true
     else
       kegs, casks = args.named.to_kegs_to_casks
 
-      kegs.each { |keg| PrettyListing.new keg } if kegs.present?
-      list_casks(casks, args: args) if casks.present?
+      if args.verbose? || !$stdout.tty?
+        if kegs.present?
+          system_command! "find", args:         kegs.map(&:to_s) + %w[-not -type d -print],
+                                  print_stdout: true
+        end
+        if casks.present?
+          system_command! "find", args:         casks.map(&:caskroom_path) + %w[-not -type d -print],
```

Comment:
Would be nice to share these `find` args.

Reply:
Also added skipping of .DS_Store files.